src/Grid/topology.jl

```diff
@@ -0,0 +1,341 @@
+
+############
+# Topology #
+############
+
+"""
+    getneighborhood(topology, grid::AbstractGrid, cellidx::CellIndex, include_self=false)
+    getneighborhood(topology, grid::AbstractGrid, faceidx::FaceIndex, include_self=false)
+    getneighborhood(topology, grid::AbstractGrid, vertexidx::VertexIndex, include_self=false)
+    getneighborhood(topology, grid::AbstractGrid, edgeidx::EdgeIndex, include_self=false)
+
+Returns all connected entities of the same type as defined by the respective topology. If `include_self` is true,
+the given entity is included in the returned list as well.
+"""
+getneighborhood
+
+struct EntityNeighborhood{T<:Union{BoundaryIndex,CellIndex}}
+    neighbor_info::Vector{T}
+end
+
+EntityNeighborhood(info::T) where T <: BoundaryIndex = EntityNeighborhood([info])
+Base.zero(::Type{EntityNeighborhood{T}}) where T = EntityNeighborhood(T[])
+Base.zero(::Type{EntityNeighborhood}) = EntityNeighborhood(BoundaryIndex[])
+Base.length(n::EntityNeighborhood) = length(n.neighbor_info)
+Base.getindex(n::EntityNeighborhood,i) = getindex(n.neighbor_info,i)
+Base.firstindex(n::EntityNeighborhood) = 1
+Base.lastindex(n::EntityNeighborhood) = length(n.neighbor_info)
+Base.:(==)(n1::EntityNeighborhood, n2::EntityNeighborhood) = n1.neighbor_info == n2.neighbor_info
+Base.iterate(n::EntityNeighborhood, state=1) = iterate(n.neighbor_info,state)
+
+function Base.:+(n1::EntityNeighborhood{T1}, n2::EntityNeighborhood{T2}) where {T1 <:Union{BoundaryIndex,CellIndex}, T2 <:Union{BoundaryIndex,CellIndex}}
+    neighbor_info = [n1.neighbor_info; n2.neighbor_info]
+    return EntityNeighborhood(neighbor_info)
+end
+
+function Base.show(io::IO, ::MIME"text/plain", n::EntityNeighborhood)
+    if length(n) == 0
+        println(io, "No EntityNeighborhood")
+    elseif length(n) == 1
+        println(io, "$(n.neighbor_info[1])")
+    else
+        println(io, "$(n.neighbor_info...)")
+    end
+end
+
+abstract type AbstractTopology end
+
+"""
+    ExclusiveTopology(cells::Vector{C}) where C <: AbstractCell
```

Comment:
docstrings are outdated and more important there should be a note that mixed dimensional connectivity is now not working. Previously it was working to some degree in the sense of dispatched `edges,faces,vertices`

Reply:
Good catch.